### PR TITLE
#840 Add tests for StreamApp APIs

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #840 )
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-streams/src/main/java/com/island/ohara/streams/data/Stele.java [hunk 1]
```diff
@@ -18,23 +18,22 @@
...
 
-@SuppressWarnings("serial")
 public final class Stele extends Data implements Serializable {
+  private static final long serialVersionUID = 1L;
   private final String kind;
   private final String key;
   private final String name;
-  private final List<String> from;
-  private final List<String> to;
+  private final String[] from;
```
Comment (chia7712) [inline ohara-streams/src/main/java/com/island/ohara/streams/data/Stele.java]: why?

File: ohara-streams/src/test/java/com/island/ohara/streams/TestStreamApp.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * limitations under the License.
+ */
+
+package com.island.ohara.streams;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestStreamApp {
```
Comment (chia7712) [inline ohara-streams/src/test/java/com/island/ohara/streams/TestStreamApp.java]: It need to extend one of our test catalog (such as SmallTest and MediumTest)

File: ohara-streams/src/test/java/com/island/ohara/streams/TestStreamAppTopology.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import com.island.ohara.streams.data.Poneglyph;
+import com.island.ohara.streams.data.Stele;
+import com.island.ohara.streams.ostream.KeyValue;
+import com.island.ohara.streams.ostream.Serdes;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestStreamAppTopology {
```
Comment (chia7712) [inline ohara-streams/src/test/java/com/island/ohara/streams/TestStreamAppTopology.java]: ditto

File: ohara-streams/src/main/java/com/island/ohara/streams/data/Stele.java (hunk 1 above)
Comment (saivirtue) [reply ohara-streams/src/main/java/com/island/ohara/streams/data/Stele.java]: We need to use the `ToStringBuilder.reflectionToString` to generate json string, and there is a wield result between `List<String>` and `String[]`
ex : val list = ("a","b")
List<String> : [a,b]
String[] : ["a", "b"]  ,== correct json object

Maybe there is other solution?

File: ohara-streams/src/test/java/com/island/ohara/streams/TestStreamApp.java (hunk 2 above)
Comment (saivirtue) [reply ohara-streams/src/test/java/com/island/ohara/streams/TestStreamApp.java]: OK

File: ohara-streams/src/main/java/com/island/ohara/streams/data/Stele.java (hunk 1 above)
Comment (chia7712) [reply ohara-streams/src/main/java/com/island/ohara/streams/data/Stele.java]: Seems a bug in Commons lang. Please file a issue for it and then document it in your code.

Comment (saivirtue) [reply ohara-streams/src/main/java/com/island/ohara/streams/data/Stele.java]: After a deep trace of `ToStringBuilder`, I think the problem is not related to commons-lang.
detail is in the comment of source code.

Comment (chia7712) [reply ohara-streams/src/main/java/com/island/ohara/streams/data/Stele.java]: I'm ok to use array type. Just left a issue for it and document for your code. It won't cause end of world.

Comment (saivirtue) [reply ohara-streams/src/main/java/com/island/ohara/streams/data/Stele.java]: see #870

**Pull request merged**